pkg/gateway/doc.go

```diff
@@ -0,0 +1,2 @@
+// Package gateway provides implementations to send data through the transport (e.g. headers).
+package gateway
```

Comment:
i don't know why we called this gateway (probably because of https://github.com/grpc-ecosystem/grpc-gateway?). But this word means nothing to me personally and i would like to rename this to `transport`.